dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -448,74 +454,441 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInSecondLanguage(new ExpectedResults(false, false))
+                .createTestCaseSiteSearch();
+
+        /*
+         * Given sceneario: two-language content referenced from a page in default lang only. DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run including both languages
+         * Expected: searching content of either version of the content should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case5 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .createContentInDefaultLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInDefaultLanguage(new ExpectedResults(true, false))
+                .expectedResultsWhenSearchingContentInSecondLanguage(new ExpectedResults(true, false))
+                .createTestCaseSiteSearch();
+
+
+
+        return new TestCaseSiteSearch[] {case1, case2, case3, case4, case5};
+    }
+
+    @UseDataProvider("siteSearchTestCases")
+    @Test
+    public void testSiteSearchDifferentScenarios(final TestCaseSiteSearch testCase)
+            throws DotPublishingException, JobExecutionException, DotDataException, IOException, DotSecurityException, WebAssetException {
+
+        boolean defaultContentToDefaultLangOriginalValue =
+                Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
+
+        boolean defaultPagetoDefaultLangOriginalValue =
+                Config.getBooleanProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE", true);
+
+        try {
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", testCase.defaultContentToDefaultLanguage);
+            Config.setProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE", testCase.defaultPageToDefaultLanguage);
+
+            final Host site = new SiteDataGen().nextPersisted();
+            Language defaultLang = APILocator.getLanguageAPI().getDefaultLanguage();
+            Language secondLang = new LanguageDataGen().nextPersisted();
+            folder = new FolderDataGen().site(site).nextPersisted();
+
+            Contentlet contentletDefaultLang = null;
+            Contentlet contentSecondLang = null;
+
+            if(testCase.createContentInDefaultLanguage) {
+                contentletDefaultLang = createAndPublishEmployeeContent(site, defaultLang, "catherine");
+
+                if(testCase.createContentInSecondLanguage) {
+                    createNewVersionAndPublishExistingEmployeeContent(secondLang, contentletDefaultLang,
+                            "catalina");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "catalina" appears 8 times in this file; the first occurrence is on line 613](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)